client/client.go

```diff
@@ -564,6 +559,11 @@ type (
 		SignalWithStartWorkflow(ctx context.Context, workflowID string, signalName string, signalArg interface{},
 			options StartWorkflowOptions, workflow interface{}, workflowArgs ...interface{}) (WorkflowRun, error)
 
+		// NewWithStartWorkflowOperation returns a WithStartWorkflowOperation to perform Update-with-Start.
+		// Returns an error if the WorkflowIDConflictPolicy is not set, or if the workflow or arguments are invalid.
+		// NOTE: Experimental
+		NewWithStartWorkflowOperation(options StartWorkflowOptions, workflow interface{}, args ...interface{}) WithStartWorkflowOperation
```

Comment:
```suggestion
		NewWithStartWorkflowOperation(options WithStartWorkflowOperationOptions) WithStartWorkflowOperation
```

Arguably this is more future proof, but understood if we don't want it

Reply:
This seems to be going too far to me: it doesn't seem right to have to create a  `WithStartWorkflowOperationOptions` struct in order to create a `WithStartWorkflowOperation` given that the latter essentially is just a stateful version of the former.

Reply:
> which disables the ability to discuss further

Apparently not :)

>  I am ok if we all agree this isn't needed

Indeed, the consensus with others was that it wasn't needed.

> I fear we will add some server-side option that only applies to multiop start workflow and we won't be able to represent it here.

We would be able to add such an option to `UpdateWithStartWorkflowOptions`.

Reply:
> We would be able to add such an option to UpdateWithStartWorkflowOptions.

Until we use this `WithStartWorkflowOperation` for more than just update with start, which is why I figured it was named to not be related to update. But it's not a big deal.

Reply:
I might be missing something but if we

> add some server-side option that only applies to multiop start workflow

we would at least be able to add as a top-level field here:

```go
type UpdateWithStartWorkflowOptions struct {
	StartWorkflowOperation WithStartWorkflowOperation
	UpdateOptions          UpdateWorkflowOptions
}
```